insalan/payment/models.py

```diff
@@ -0,0 +1,369 @@
+"""Payment Models"""
+from decimal import Decimal
+from datetime import datetime
+import itertools
+import logging
+import uuid
+
+import requests
+
+from django.db import models
+from django.utils.translation import gettext_lazy as _
+from django.utils import timezone
+from rest_framework.serializers import ValidationError
+
+import insalan.settings as app_settings
+
+from insalan.tournament.models import Tournament
+from insalan.user.models import User
+
+from .tokens import Token
+
+logger = logging.getLogger(__name__)
+
+
+class TransactionStatus(models.TextChoices):
+    """Information about the current transaction status"""
+
+    FAILED = "FAILED", _("échouée")
+    SUCCEEDED = "SUCCEEDED", _("Réussie")
+    PENDING = "PENDING", _("En attente")
+    REFUNDED = "REFUNDED", _("Remboursé")
+
+
+class ProductCategory(models.TextChoices):
+    """Different recognized categories of products"""
+
+    REGISTRATION_PLAYER = "PLAYER_REG", _("Inscription joueur⋅euse")
+    REGISTRATION_MANAGER = "MANAGER_REG", _("Inscription manager")
+    PIZZA = "PIZZA", _("Pizza")
+
+
+class Product(models.Model):
+    """Object to represent in database anything sellable"""
+
+    price = models.DecimalField(
+        null=False, max_digits=5, decimal_places=2, verbose_name=_("prix")
+    )
+    name = models.CharField(max_length=50, verbose_name=_("intitulé"))
+    desc = models.CharField(max_length=50, verbose_name=_("description"))
+    category = models.CharField(
+        max_length=20,
+        blank=False,
+        null=False,
+        verbose_name=_("Catégorie de produit"),
+        default=ProductCategory.PIZZA,
+        choices=ProductCategory.choices,
+    )
+    associated_tournament = models.ForeignKey(
```

Comment:
do we keep this field or use a more generic foreignKey?

Reply:
I could try and work out how to do a more generic foreign key now to be honest

I'm just worried about breaking things and not being able to fix them before merge, but I can try and check if it's feasible at least

Reply:
as you want, you did a huge work here